src/query/dblink_scan.c

```diff
@@ -570,7 +601,17 @@ dblink_execute_query (struct access_spec_node *spec, VAL_DESCR * vd, DBLINK_HOST
     }
   else
     {
-      cci_set_autocommit (conn_handle, CCI_AUTOCOMMIT_TRUE);
+      if (auto_commit)
+	{
+	  spec->s.dblink_node.conn_handle = -1;
+	  cci_set_autocommit (conn_handle, CCI_AUTOCOMMIT_TRUE);
+	}
+      else
+	{
+	  spec->s.dblink_node.conn_handle = conn_handle;
+	  cci_set_autocommit (conn_handle, CCI_AUTOCOMMIT_FALSE);
+	}
+
```

Comment:
```suggestion
	  spec->s.dblink_node.conn_handle = auto_commit ? -1 : conn_handle;
	  cci_set_autocommit (conn_handle, auto_commit ? CCI_AUTOCOMMIT_TRUE : CCI_AUTOCOMMIT_FALSE);
```
다음과 같이 간단하게 작성할 수도 있을 거 같습니다.